testsuite/tests/misc/sort_segment.ml

```diff
@@ -0,0 +1,47 @@
+(* TEST *)
+
+open Printf
+
+(* [generate n] generates an array of length [n],
+   containing random integer data. *)
+let generate n =
+  Array.init n @@ fun i -> Random.int n
+
+(* Our comparison function. *)
+let cmp =
+  Int.compare
+
+(* Comparing the content of two arrays. *)
+let equal a1 a2 =
+  assert (Array.length a1 = Array.length a2);
+  Array.for_all2 (=) a1 a2
+
+(* [test a ofs len] tests [Array.stable_sort_segment a ofs len].
+   [Array.sort] is used as a reference sorting algorithm. *)
+let test a ofs len =
+  let segment = Array.sub a ofs len in
+  Array.sort cmp segment;
+  let expected =
+    Array.concat [
+      Array.sub a 0 ofs;
+      segment;
+      Array.sub a (ofs+len) (Array.length a - (ofs+len))
+    ]
+  in
+  Array.stable_sort_segment cmp a ofs len;
+  if not (equal a expected) then
+    printf "Array.stable_sort_segment: FAILURE (array length %d, offset %d, segment length %d)\n%!"
+      (Array.length a) ofs len
+
+(* The main loop. *)
+let number_of_tests = 5000
+let max_length = 128
```

Comment:
We could also decide to exhaustively test all segments on a size-10 arrays, which may provide similar test coverage (it is not robust to algorithms using larger cutoffs, but luckily the sort code uses `cutoff = 5`) at a fraction of the compute cost -- there are 45 such segments, total length 210.

Reply:
OK. I have reduced `number_of_tests` from 5000 to 1000 and have added tests of the kind that you suggest.